### PR TITLE
Bind SDK ChipError and ErrorStr

### PR DESCRIPTION
**Pull request by krypton36**
#### Problem
* Binding ErrorStr and ChipError

#### Change overview
* Updated import style
* Update error bindings to use SDK directly

#### Testing
```python
from pychip import chip
e = chip.ChipError(50)
chip.ErrorStr(e)
```

### CODE REVIEW

File: src/pybindings/pycontroller/ControllerBindings/PyChip_ChipError.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#include <core/CHIPError.h>
+#include <sstream> // __str__
```
Comment (andy31415) [inline src/pybindings/pycontroller/ControllerBindings/PyChip_ChipError.cpp:2]: what does the comment `__str__` mean here and below? It does not seem to help me understand the code.

File: src/pybindings/pycontroller/ControllerBindings/PyChip_ChipError.cpp [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+#ifndef BINDER_PYBIND11_TYPE_CASTER
+#define BINDER_PYBIND11_TYPE_CASTER
+PYBIND11_DECLARE_HOLDER_TYPE(T, std::shared_ptr<T>)
+PYBIND11_DECLARE_HOLDER_TYPE(T, T *)
+PYBIND11_MAKE_OPAQUE(std::shared_ptr<void>)
+#endif
+
+void bind_PyChip_ChipError(std::function<pybind11::module &(std::string const & namespace_)> & M)
+{
+    { // chip::ChipError file:core/CHIPError.h line:50
```
Comment (andy31415) [inline src/pybindings/pycontroller/ControllerBindings/PyChip_ChipError.cpp:18]: remove this comment: line may become obsolete in time if chiperror moves. same below.

I am working here on the assumption that these are maintained manually. If these are generated files, could we make generation part of the build process instead or does that require too much 3rd party?

**Review by andy31415 — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

File: src/pybindings/pycontroller/ControllerBindings/PyChip_ChipError.cpp (hunk 2 above)
Comment (krypton36) [reply src/pybindings/pycontroller/ControllerBindings/PyChip_ChipError.cpp:18]: I will remove the comments. This is a semi-generated file, but will require compiling binder and manuals intervention as we do not want all SDK apis bound.

File: src/pybindings/pycontroller/ControllerBindings/PyChip_ChipError.cpp (hunk 1 above)
Comment (krypton36) [reply src/pybindings/pycontroller/ControllerBindings/PyChip_ChipError.cpp:2]: Will remove

**Review by chrisdecenzo — APPROVED** [review]

**Pull request merged**